src/main/java/io/fabric8/jenkins/openshiftsync/SecretWatcher.java

```diff
@@ -163,16 +163,14 @@ public void eventReceived(Action action, Secret secret) {
     private void upsertCredential(final Secret secret) throws Exception {
         if (validSecret(secret)) {
             CredentialsUtils.upsertCredential(secret);
-            trackedSecrets.put(secret.getMetadata().getUid(), secret
-                    .getMetadata().getResourceVersion());
+            trackedSecrets.put(secret.getMetadata().getUid(), secret.getMetadata().getResourceVersion());
```

Comment:
I can only see trackedSecrets.put or trackedSecrets.remove , but I never see a trackedSecrets.get ?

Reply:
If the tracked secret already exists it is updated by overriding it with put.